tools/omnissm/inventory/process/process.go

```diff
@@ -0,0 +1,112 @@
+/*
+Copyright 2018 Capital One Services, LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+
+You may obtain a copy of the License at
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/shirou/gopsutil/process"
+)
+
+func main() {
+
+	procs, _ := process.Pids()
+	procInfos := make([]map[string]interface{}, 0)
+
+	for _, pid := range procs {
+		procMap := map[string]interface{}{}
+		proc, err := process.NewProcess(int32(pid))
+		if err != nil {
+			continue
+		}
+		procMap["process_pid"] = int32(proc.Pid)
+
+		name, err := proc.Name()
+		if err == nil {
+			procMap["process_name"] = name
+		} else {
+			fmt.Println("Error name", pid, err)
+		}
+
+		user, err := proc.Username()
+		procMap["process_user"] = user
+
+		/* A bit expensive for busy systems.
+
+		// ipv4 only, capture listen ports, requires all conn enum.
+			conns, err := proc.Connections()
+			if err == nil {
+				ports := make([]uint32, 0)
+
+				for _, conn := range conns {
+					if conn.Status != "LISTEN" {
+						continue
+					}
+					ports = append(ports, conn.Laddr.Port)
+				}
+				if len(ports) > 0 {
+					procMap["listen_ports"] = ports
+				}
+			}
+		*/
+
+		cmdline, err := proc.Cmdline()
+		if err == nil {
```

Comment:
Branch here should short circuit on `err != nil` instead of using the else.  Also should this return from the function when it encounters an error?  It currently prints the error and continues.